cypress/README.md

```diff
@@ -0,0 +1,603 @@
+
+To keep it DRY we can create reusable data-e2e locators. For example, we have a lot of buttons on the page. We can create a reusable data-e2e locator for all buttons. And then use it in our tests.
+
+```html
+<button data-e2e="COMMON_BUTTON__FIRST">First</button>
+<button data-e2e="COMMON_BUTTON__SECOND">Second</button>
+<button data-e2e="COMMON_BUTTON__THIRD">Third</button>
+```
+Then we need to create a selector and a get method that will be replaced `{buttonName}` text in selector to need text.
+
+```typescript
+const commonSelectors = {
+    button: '[data-e2e="COMMON_BUTTON__{buttonName}"]'
+}
+```
+
+```typescript
+    public getButtonName(buttonName: string): string {
+        return commonSelectors.button.replace('{buttonName}', buttonName).toUpperCase();
+    }
+```
+
+In the end we get `data-e2e="COMMON_BUTTON` selector with the needed name.
+
+For example:
+
+```typescript
+basePage.getButtonName('First') // will return us `data-e2e="COMMON_BUTTON__FIRST`
+```
+
+And we can use it an argument for the check or click etc. methods
+
+```typescript
+basePage.isElementExist({
+    selector: basePage.getButtonName('First')
+})
+```
+
+### Skip tests with `skipWhen()` method
+
+Sometimes you need to skip some tests. For example, you need to skip tests for the sample that don't have such elements on the page or even have another bussiness logic in it. For that case you can use `skipWhen()` method.
+
+Note: This method was added to Cypress by custom command functionality. More about adding custom commans [here](https://docs.cypress.io/api/cypress-api/custom-commands)
+
+Usage:
+
+🤔 Imagine you have two apps with one button but button in the first app create the user and button in the second app delete the user and that buttons exist on separate hosts. So you write the tests to check creating the user and deleting the user. It looks like:
+
+```typescript
+const appData = [
+    {
+        host: 3001
+    },
+    {
+        host: 3002
+    }
+]
+
+appsData.forEach(
+    (property: {
+        host: number
+    }) => {
+        describe(`Check example apps`, () => {
+            beforeEach(() => {
+                basePage.openLocalhost(host)
+            })
+
+            it(`User should be created & added`, () => {
+                basePage.clickElementBySelector({
+                    selector: baseSelectors.createButton
+                })
+                basePage.checkUserCreated(true)
+            })
+
+            it(`User should be deleted`, () => {
+                basePage.clickElementBySelector({
+                    selector: baseSelectors.deleteButton
+                })
+                basePage.checkUserDeleted(true)
+            })
+        })
+    })
+```
+
+It will be run your testCases on every host, as example on 3001 and 3002. So you need to skip delete user on 3001 host and creating user on 3002. Let's do this!
+
+First of all you need to add custom command to `cypress/support/commands.ts`:
```

Comment:
A little bit confusing. Command already added for us. Better to rephrase. Remake it to something like .. check how we already added skipWhen and keep in mind that you can add any custom command as simple as that if you need so

Reply:
Fixed